src/protocol/modulus_conversion/convert_shares.rs

```diff
@@ -27,33 +47,49 @@ impl AsRef<str> for Step {
     }
 }
 
-/// Internal use only.
+/// Convert one bit of an XOR sharing into a triple of replicated sharings of that bit.
+/// This is not a usable construct, but it can be used with `convert_one_bit` to produce
+/// a single replicated sharing of that bit.
+///
 /// This is an implementation of "Algorithm 3" from <https://eprint.iacr.org/2018/387.pdf>
 ///
-fn local_secret_share<B: BinaryField, F: Field>(
-    input: &Replicated<B>,
+pub fn convert_bit_local<F: Field>(
     helper_role: Role,
+    bit_index: u32,
+    input: &XorReplicated,
 ) -> [Replicated<F>; 3] {
-    let (left, right) = input.as_tuple();
+    let left = u128::from(input.left() >> bit_index) & 1;
+    let right = u128::from(input.right() >> bit_index) & 1;
     match helper_role {
         Role::H1 => [
-            Replicated::new(F::from(left.as_u128()), F::ZERO),
-            Replicated::new(F::ZERO, F::from(right.as_u128())),
+            Replicated::new(F::from(left), F::ZERO),
+            Replicated::new(F::ZERO, F::from(right)),
             Replicated::new(F::ZERO, F::ZERO),
         ],
         Role::H2 => [
             Replicated::new(F::ZERO, F::ZERO),
-            Replicated::new(F::from(left.as_u128()), F::ZERO),
-            Replicated::new(F::ZERO, F::from(right.as_u128())),
+            Replicated::new(F::from(left), F::ZERO),
+            Replicated::new(F::ZERO, F::from(right)),
         ],
         Role::H3 => [
-            Replicated::new(F::ZERO, F::from(right.as_u128())),
+            Replicated::new(F::ZERO, F::from(right)),
             Replicated::new(F::ZERO, F::ZERO),
-            Replicated::new(F::from(left.as_u128()), F::ZERO),
+            Replicated::new(F::from(left), F::ZERO),
         ],
     }
 }
 
+pub fn convert_bits_local<F: Field>(
```

Comment:
Unclear naming. "Convert Bits" sounds like it could indicate converting all the bits of a number.

Perhaps `convert_bit_local_for_list` or `convert_bit_local_for_all`